src/Conversion/ONNXToKrnl/Tensor/Gather.cpp

```diff
@@ -64,15 +64,16 @@ struct ONNXGatherOpLowering : public ConversionPattern {
     int jIndexStart = iIndexStart + axisLit;
     int kIndexStart = jIndexStart + indicesRank - (axisLit + 1);
 
-    // Load the constants for the tests
+    // Insert code inside the loop.
+    rewriter.setInsertionPointToStart(outputLoops.getIterateBlock());
+    IndexExprScope innerLoopScope;
```

Comment:
No usage of innerLoopScope?

Reply:
@chentong319  

That is the new use; scopes are used implicitly, just like insertion points. They use a thread private pointers, and so do I now.

Reply:
Can we have a better way, such as a static method? To me, it is against intuition to just declare a local variable.

Reply:
No, it need to be a variable; when created, it add itself to the top context, and when removed, it reinstall the parent context.

It is the same pattern as the EDSC scope.